### PR TITLE
Finished backend Resource tests

### PR DESCRIPTION
**Pull request by alicesf2**
## Status:

:rocket: Ready

## Description

Fixes #<number>

### CODE REVIEW

File: backend/test/resources.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+require("./auth_stubs").stubAllAuth();
+const app = require("../app.js");
+const Resource = require("../models/Resource");
+
+const sampleResourceInfo = {
+  companyName: "Google",
+  contactName: "Alice",
+  contactPhone: "123456789",
+  contactEmail: "123@gmail.com",
+  description: "Let me dream",
```
Comment (josh-byster) [inline backend/test/resources.js]: LOL 👍

File: backend/test/resources.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  },
+  tags: ["social"]
+};
+
+const createSampleResource = async (resourceInfo = sampleResourceInfo) => {
+  const newResource = new Resource(resourceInfo);
+  await newResource.save();
+};
+
+const createSampleResource2 = async (resourceInfo = sampleResourceInfo2) => {
```
Comment (josh-byster) [inline backend/test/resources.js]: Can we merge this method with the one above it, and just have us pass in the alternative `sampleResourceInfo2` if we want to use that one instead? This way we don't have repeated code.

File: backend/test/resources.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+beforeEach(() => createSampleResource());
+afterEach(() => Resource.remove({}));
+
+describe("GET /resources", () => {
+  beforeEach(() => Resource.remove());
+  it("should get no Resources", async () => {
+    const res = await request(app)
+      .get(`/api/resources`)
+      .expect(200);
+    expect(res.body.result).to.be.an("array").that.is.empty;
```
Comment (josh-byster) [inline backend/test/resources.js]: Well done using Chai asserts properly :) 👍

File: backend/test/resources.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  const newResource = new Resource(resourceInfo);
+  await newResource.save();
+};
+
+beforeEach(() => createSampleResource());
+afterEach(() => Resource.remove({}));
+
+describe("GET /resources", () => {
+  beforeEach(() => Resource.remove());
+  it("should get no Resources", async () => {
```
Comment (josh-byster) [inline backend/test/resources.js]: Awesome work testing out the edge cases like this!

File: backend/test/resources.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+});
+
+describe("GET /resources/:resource_id", () => {
+  it("should get a Resource by id", async () => {
+    const resource = await Resource.findOne({ companyName: "Google" });
+    const resourceId = resource._id;
+    const res = await request(app)
+      .get(`/api/resources/${resourceId}`)
+      .expect(200);
+    expect(res.body.result.contactName).equals("Alice");
```
Comment (josh-byster) [inline backend/test/resources.js]: Great flow on testing this endpoint out :)

File: backend/test/resources.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+  const newResource = new Resource(resourceInfo);
+  await newResource.save();
+};
+
+const createSampleResource2 = async (resourceInfo = sampleResourceInfo2) => {
+  const newResource = new Resource(resourceInfo);
+  await newResource.save();
+};
+
+beforeEach(() => createSampleResource());
```
Comment (josh-byster) [inline backend/test/resources.js]: Nit: What do you think about removing this and adding this to individual `describe` blocks when necessary? This way upon reaching each `describe` block, have a clean slate in our database to work with.
